legend-depot-core-data-services/src/main/java/org/finos/legend/depot/services/projects/ProjectsServiceImpl.java

```diff
@@ -269,22 +268,17 @@ public ProjectDependencyReport getProjectDependencyReport(List<ProjectVersion> p
         ProjectDependencyGraph graph = new ProjectDependencyGraph();
         ProjectDependencyGraphWalkerContext  graphWalkerContext =  new ProjectDependencyGraphWalkerContext();
         buildDependencyGraph(graph, null, projectDependencyVersions, graphWalkerContext);
-        ProjectDependencyReport report = buildReportFromGraph(graph, graphWalkerContext);
-        return overrideConflictDependencies(projectDependencyVersions, report);
+        buildProjectVersionMap(projectDependencyVersions, graphWalkerContext);
+        return buildReportFromGraph(graph, graphWalkerContext);
     }
 
-    public ProjectDependencyReport overrideConflictDependencies(List<ProjectVersion> projectDependencyVersions, ProjectDependencyReport report)
+    private void buildProjectVersionMap(List<ProjectVersion> projectDependencyVersions, ProjectDependencyGraphWalkerContext graphWalkerContext)
     {
-        List<ProjectDependencyReport.ProjectDependencyConflict> conflicts = new ArrayList<>(report.getConflicts());
-        projectDependencyVersions.stream().forEach(dep ->
+        projectDependencyVersions.forEach(pv ->
         {
-            Optional<ProjectDependencyReport.ProjectDependencyConflict> conflictPresent = conflicts.stream().filter(conflict -> conflict.getGroupId().equals(dep.getGroupId()) && conflict.getArtifactId().equals(dep.getArtifactId())).findFirst();
-            if (conflictPresent.isPresent())
-            {
-                report.removeConflict(conflictPresent.get());
-            }
+            StoreProjectVersionData versionData = graphWalkerContext.getProjectData(pv.getGroupId(), pv.getArtifactId(), pv.getVersionId());
+            versionData.getTransitiveDependenciesReport().getTransitiveDependencies().forEach(dep -> graphWalkerContext.addVersionToProject(dep.getGroupId(), dep.getArtifactId(), dep));
```

Comment:
trying to understand this a bit. Are we ensuring that the transitive depdency of each project has already had the override occur and when we get its transitive dependency only one version of that dependency will occur. 

say we have projectA v1 with the following dependency tree
`
project B V1
    project Z v1
Project C V1
    project Z V2 
Project Z V2
`

if i query for the dependency (including transitive) will i get get project z v2 or both? 
Would be good if we had the above as a test case :)

Reply:
It would have the full tree mentioned but while calculating conflicts it would have the overridden dependencies, so Project B V1, Project C V1 and Project Z V2, since transitive dependencies become our main source of truth.

There is  an already added test case which showcases this behavior when we did the earlier change, this is to support deps which have had overriding done and they ae being included in the project